### ISSUE TITLE
JetSmartFilter Set Default Filter selection without using querystring

### ISSUE THREAD

**Issue by ingramleedy** [CrocoBlock/suggestions#2548]
*Set Default Filter selection without using querystring.    Ability to set in the widget whether its selected by first initial state.

**Comment by hum-bug** [comment]
Looking for the same - should be an option for a default filter value.

**Comment by adelaiglesia** [comment]
Push to this. I'm writing javascript just for this. A filter system should have default values and limits. And the (expected) obfuscation of jetsmartfilters javascript is doing things hard for us.

**Comment by SornyKonst** [comment]
This is a very important but unfortunately missing feature. URL query string is not an option in many cases. How can such a simple function be missing in such a great plugin? JS is also not a solution in many cases, because it is only applied after the unfiltered CPT query has been loaded or after caching plugins have indexed the page. Perhaps the support could also provide a PHP code to load a certain filter value as default for a certain page ID. I think an option for a default active filter should not be missing in the next update.

**Comment by marowi85** [comment]
I miss that feature too. Would be a great addition.

**Comment by rquirola** [comment]
+1

**Comment by heikodietze** [comment]
Absolutely needed! e.g. I've got a clients restaurant website with a menu filterable by weekdays. Initially now it shows the menus of all weekdays, which is stupid and not a solution that I can really explain to the client...

**Comment by kskonovalov** [comment]
+1 for this

**Comment by stasonua0** [comment]
A very popular feature, I ask for more attention to this request from the developers

**Comment by lux-angelique** [comment]
I yearn for this feature! That would be so handy.

**Comment by stasonua0** [comment]
Here is an example of solving the problem

`<script>
jQuery(window).on('load', function() {
      jQuery(".jet-color-image-list__item input[data-label='Курсы английского языка']").click();
    });
</script>`

**Comment by kimradford** [comment]
This would be amazing!

**Comment by Tygovanommen** [comment]
+1

**Comment by vmavrikios** [comment]
+1 indeed
This way you can manage archive pages for unique combinations of characteristics
Any news on this..?

**Comment by Smalyleinchen** [comment]
> Here is an example of solving the problem
> 
> `<script> jQuery(window).on('load', function() { jQuery(".jet-color-image-list__item input[data-label='Курсы английского языка']").click(); }); </script>`

Thx for sharing. That helped me a lot :)

**Comment by Smalyleinchen** [comment]
+1 Absolutely needed

**Comment by shahinshenavary** [comment]
https://crocoblock.com/knowledge-base/articles/jetsmartfilters-create-predefined-filter-results-with-dynamic-tag/

**Comment by rlb222** [comment]
+1 without use of a button. I want to enter the SmartFilter-search-page and prefill some Smartfilter fields with data from the current users meta-box.
thx!

**Comment by mateitudor** [comment]
> Here is an example of solving the problem
> 
> `<script> jQuery(window).on('load', function() { jQuery(".jet-color-image-list__item input[data-label='Курсы английского языка']").click(); }); </script>`

This is a really good fix, too bad it flashes "All" before it applies the filter. :(

**Comment by pajtai** [comment]
@mateitudor I have to agree. The pause between the initial state and the updated state can be quite lengthy, so the jQuery fix really isn't practical.

**Comment by kskonovalov** [comment]
> This is a really good fix, too bad it flashes "All" before it applies the filter. :(

you can maybe hide it with css and after it's loaded show it back. like this:
1. add class to the filters like .filters-wrapper (or use existing)
2. add to css .filters-wrapper {display: none;}
3. change the code to
`<script> jQuery(window).on('load', function() { jQuery(".jet-color-image-list__item input[data-label='Курсы английского языка']").click(); jQuery(".filters-wrapper").show(100); }); </script>`

**Comment by quirkautogroupmaine** [comment]
Thank you for sharing this tip. I placed this code in the bottom of the body in an Elementor template and it's not working. Does it need to be in the page header?

**Comment by kskonovalov** [comment]
> Thank you for sharing this tip. I placed this code in the bottom of the body in an Elementor template and it's not working. Does it need to be in the page header?

you should adjust the code for your case, at least check the data-label=['...'] selector, but better check all selectors

**Comment by serdarozbilen** [comment]
Same here. +1

**Comment by scottamarkham** [comment]
+1, this also needs to allow the default filter to be dynamic, e.g. select first value in dropdown list, select current year, etc

**Comment by pa2codes** [comment]
This feature is needed, please add it :) thx

**Comment by patrickadam** [comment]
> Here is an example of solving the problem
> 
> `<script> jQuery(window).on('load', function() { jQuery(".jet-color-image-list__item input[data-label='Курсы английского языка']").click(); }); </script>`

Unfortunately this doesn't seem to work anymore. I've tried with the select option, but no change.

**Comment by lcrvtt** [comment]
+1

**Comment by raulopezvi** [comment]
> > Here is an example of solving the problem
> > `<script> jQuery(window).on('load', function() { jQuery(".jet-color-image-list__item input[data-label='Курсы английского языка']").click(); }); </script>`
> 
> Unfortunately this doesn't seem to work anymore. I've tried with the select option, but no change.

19 May 2023. This solution works perfectly!

**Comment by maxwh1** [comment]
+1 to adding this feature please

**Comment by soiluoi** [comment]
I don't know why but click() trick is not working for now. it only work one time. after that it just click but not trigger change event. so you just make it click and change query builder like this. set fallback. and it will be a first state.
![image](https://github.com/Crocoblock/suggestions/assets/37627736/7148ad2e-40fe-4d4b-bc65-5c7e4bf33f24)

**Comment by OmniasDigital** [comment]
> I don't know why but click() trick is not working for now. it only work one time. after that it just click but not trigger change event. so you just make it click and change query builder like this. set fallback. and it will be a first state. ![image](https://user-images.githubusercontent.com/37627736/244809083-7148ad2e-40fe-4d4b-bc65-5c7e4bf33f24.png)

What query type is this?

**Comment by portrayaloflife** [comment]
+1

**Comment by mamasakhlisi27** [comment]
<script>
  addEventListener("DOMContentLoaded", (event) => {
    setTimeout(() => {
      let element = document.querySelector("input[data-label='South Africa']");

      if (element) {
        element.click()
      } 
    }, 1000);
  });
</script>

**Comment by alephjericho** [comment]
+1

**Comment by duyle96** [comment]
I wrote this and it's working :)
```

// Find the label with the text 'Nejhledanější'
    $('.jet-checkboxes-list__label').each(function() {
        if ($(this).text().trim() === 'Nejhledanější') {
            // Get the parent .jet-checkboxes-list__item element
            var parentItem = $(this).closest('.jet-checkboxes-list__item');

            // Trigger a click event on it
            parentItem.trigger('click');

            // Add 'active' class to it
            parentItem.addClass('active');
        }
    });

```

**Comment by kskonovalov** [comment]
> I wrote this and it's working :)
> 
> ```
> 
> // Find the label with the text 'Nejhledanější'
>     $('.jet-checkboxes-list__label').each(function() {
>         if ($(this).text().trim() === 'Nejhledanější') {
>             // Get the parent .jet-checkboxes-list__item element
>             var parentItem = $(this).closest('.jet-checkboxes-list__item');
> 
>             // Trigger a click event on it
>             parentItem.trigger('click');
> 
>             // Add 'active' class to it
>             parentItem.addClass('active');
>         }
>     });
> ```

try to add after this line
    `$('.jet-checkboxes-list__label').each(function() {`

`console.log($(this).text().trim());`

and check the console (F12) to see what's there. Probably the encoding issue

**Comment by SargasTM** [comment]
a default filter value was added in version 3.4.0
<img width="726" alt="image" src="https://github.com/Crocoblock/suggestions/assets/5094971/0fcbb760-3f37-45cb-91cb-2dffd84f9c17">
The value will be applied to the provider once via an Ajax request when the filter is added to the page

**Comment by PhilAV** [comment]
> a default filter value was added in version 3.4.0 <img alt="image" width="726" src="https://private-user-images.githubusercontent.com/5094971/304747338-0fcbb760-3f37-45cb-91cb-2dffd84f9c17.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDc5MzIyMDUsIm5iZiI6MTcwNzkzMTkwNSwicGF0aCI6Ii81MDk0OTcxLzMwNDc0NzMzOC0wZmNiYjc2MC0zZjM3LTQ1Y2ItOTFjYi0yZGZmZDg0ZjljMTcucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDIxNCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDAyMTRUMTczMTQ1WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9YTkyMTgyNDM5MDVhMWEyMTg5MDczN2NkYjc2OTM0ZjZiMTA4YjNkZWQ4OWQ0ZWQ5ZGU0ZmRlNGQ0ZDQxZTA1OCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.FVV0qLt7uYRKGwLtaBnUzULi1TrJ0Ag7IEEkp3R6cMY"> The value will be applied to the provider once via an Ajax request when the filter is added to the page

@SargasTM , has version 3.4.0 been released? I only see 3.3.2 as the latest option available for JetSmartFilter.

**Comment by duyle96** [comment]
There is no version 3.4.0 available...

**Comment by Floydtm** [comment]
wait a little longer for the 3.4.0 release, we are working on it

**Comment by Crocoblock** [comment]
was released in v3.4.0

**Comment by erizonmacani** [comment]
My goal was to change the sorting of an Elementor loop grid. I tried with php codes modifying the query itself with no success.

Here is a work around which selects the first option on your sorting dropdown upon page load.

```
<script>
jQuery(window).on('load', function() {
    // Select the second option of the dropdown
    jQuery(".jet-sorting.jet-sorting--flex-row select option").eq(1).prop('selected', true);
    
    // Trigger change event on the dropdown
    jQuery(".jet-sorting.jet-sorting--flex-row select").change();
});

jQuery(document).ready(function($) {
    // Find the placeholder option and hide it
    $('.jet-sorting-select option:first-child').hide();
});
</script>
``` 
The nice thing is that it hides the default placeholder, making your first option appear as default.

I added it into the template within a html widget.

Hope this can help someone save some time!
